server/src/services/order.service.ts

```diff
@@ -1,12 +1,227 @@
+            e
+          );
+          return null;
+        }
+      })
+    ); 
+    const filteredStatusHistory = statusHistory.filter(
+      (result) => result !== null
+    ) as OrderStatusItemModel[];
+    return filteredStatusHistory;
+  }
+
+  private async getProductsByIds(productsEntry: string[]): Promise<ProductModel[]> {
+    const products = await Promise.all(
+      productsEntry.map(async (id) => {
+        try {
+          const product = await this.productService.getProductById(id);
+          return product;
+        } catch (e) {
+          logger.error(
+            `[OrderService][getOrderById] Error while processing product ${id}:`,
+            e
+          );
+          return null;
+        }
+      })
+    ); 
+    const filteredProdutct = products.filter(
+      (result) => result !== null
+    ) as ProductModel[];
+    return filteredProdutct;
+  }
+
+  public async getAllOrderByUserId(userId: string): Promise<OrderModel[]> {
+    try {
+      const orders = await this.orderRepository.getOrder();
+      const filteredOrdersByUser = orders.filter(
+        order => order.userId === userId
+      ) as OrderEntity[]
+      const results = await Promise.all(
+        filteredOrdersByUser.map(async (order) => {
+          try {
+            const statusHistory = await this.getOrderStatusItemsByIds(order.statusHistory);  
+            const products = await this.getProductsByIds(order.productsIds);
+            return new OrderModel({
+              ...order,
+              statusHistory: statusHistory,
+              products: products,
+              purchaseDate: new Date(order.purchaseDate)
+            });
+          } catch (e) {
+            logger.error(
+              `[OrderService][getOrders] Error while processing order ${order.id}:`,
+              e
+            );
+            return null;
+          }
+        })
+      );
+      const filteredResults = results.filter(
+        (result) => result !== null
+      ) as OrderModel[];
+      return filteredResults
+    } catch (e) {
+      throw e;
+    }
+  }
+  public async getHistoryByUserId(userId: string, historyId: any, productName: any, purchaseDate: any): Promise<OrderModel[]> {
+    try {
+      const orders: OrderModel[] = await this.getAllOrderByUserId(userId);
+      const results = await Promise.all(
+        orders.map(async (order) => {
+          try {
+            let hasStatusHistory = false;
+            let hasSomeProductWithName = false;
+            let hasPurchaseDate = false;
+  
+            if (!!purchaseDate) {
+              hasPurchaseDate = order.purchaseDate.toISOString() === purchaseDate;
+            } else {
+              hasPurchaseDate = true; // se não foi fornecido purchaseDate, assume-se que foi atendido
+            }
+  
+            if (!!historyId) {
+              hasStatusHistory = order.statusHistory.some((status) => status.id === historyId);
```

Comment:
You can do, something like that:

hasStatusHistory = order.statusHistory.some((status) => historiesStatus.includes(status.name));